### PR TITLE
improve GraphQL Mutations section

### PR DESCRIPTION
**Pull request by catalinpit**

### CODE REVIEW

File: tutorials/graphql/intro-graphql/tutorial-site/content/graphql-mutations.md [hunk 1]
```diff
@@ -72,17 +101,44 @@ mutation {
...
+        }
+      ]
+    }
+  }
+}
+```
+
+The mutation returns all the todo data we requested; nothing more, nothing less.
+
+You might ask - both GraphQL Mutations and Queries can return data? Yes, but there is a caveat. A GraphQL Mutation can only return the newly inserted data. A GraphQL Query can return multiple records.
```
Comment (praveenweb) [inline tutorials/graphql/intro-graphql/tutorial-site/content/graphql-mutations.md:130]: This is actually not true. It totally depends on the server implementation. For example, even in Hasura, we can request relationships in the response data which might be old rows.

Can we update this section to reflect that?

File: tutorials/graphql/intro-graphql/tutorial-site/content/graphql-mutations.md [hunk 2]
```diff
@@ -106,15 +164,226 @@ mutation($todo: todos_insert_input!){
...
+      user {
+        id
+        name
+      }
+    }
+  }
+}
+```
+
+https://hasura.io/learn/graphql/intro-graphql/graphql-queries/
```
Comment (praveenweb) [inline tutorials/graphql/intro-graphql/tutorial-site/content/graphql-mutations.md:287]: What is the context for this link being here?

File: tutorials/graphql/intro-graphql/tutorial-site/content/graphql-mutations.md [hunk 3]
```diff
@@ -1,66 +1,95 @@
...
+What can we observe by looking at the above mutation?
+1. It uses the keyword `mutation` rather than `query`.
+2. The mutation takes an input. We provide the title of the new todo - `Learn GraphQL`.
+3. The mutation returns data. It returns the specified fields for the newly inserted todo.
+
+You will see more GraphQL Mutations later and you will also have the chance to try them in the GraphiQL IDE.
+
+## Types of GraphQL Mutations {#types-of-graphql-mutations}
+
+There are three types of mutations in GraphQL:
```
Comment (praveenweb) [inline tutorials/graphql/intro-graphql/tutorial-site/content/graphql-mutations.md:45]: We can rephrase this to `Mutations in GraphQL can be categorised under the following types.

The reason is, GraphQL spec doesn't have `types` of mutations in particular. It is just our nomenclature.

File: tutorials/graphql/intro-graphql/tutorial-site/content/graphql-mutations.md (hunk 2 above)
Comment (catalinpit) [reply tutorials/graphql/intro-graphql/tutorial-site/content/graphql-mutations.md:287]: Added it by mistake. Will remove it

File: tutorials/graphql/intro-graphql/tutorial-site/content/graphql-mutations.md (hunk 1 above)
Comment (catalinpit) [reply tutorials/graphql/intro-graphql/tutorial-site/content/graphql-mutations.md:130]: Thanks for pointing it out.

I will remove the paragraph altogether to avoid confusion!
